Client/Frontend/Browser/BrowserViewController/BVC+Rewards.swift

```diff
@@ -115,18 +115,41 @@ extension Tab {
     func reportPageLoad(to rewards: BraveRewards) {
         guard let webView = webView, let url = webView.url else { return }
         if url.isLocal || PrivateBrowsingManager.shared.isPrivateBrowsing { return }
+                
+        var htmlBlob: String?
+        var classifierText: String?
         
-        let getHtmlToStringJSCall = "document.documentElement.outerHTML.toString()"
-        // Copy to var, as `shouldClassifyLoadsForAds` can be reset before JS completes
-        let shouldClassify = shouldClassifyLoadsForAds
-        webView.evaluateJavaScript(getHtmlToStringJSCall, completionHandler: { html, _ in
+        let group = DispatchGroup()
+        group.enter()
+        webView.evaluateJavaScript("document.documentElement.outerHTML.toString()", completionHandler: { html, _ in
+            defer { group.leave() }
             guard let htmlString = html as? String else { return }
+            htmlBlob = htmlString
```

Comment:
Can drop the guard as just use `htmlBlob = html as? String`
defer wouldn't be needed to then

Reply:
The `html` isn't a `String` though, so either we guard like this or we just do `htmlBlob = html as? String`. I can swap it to the latter if you think that's better

Reply:
It's now just one line of code more, no?